addons/website_event_meet/controllers/main.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from werkzeug.exceptions import NotFound, Forbidden
+from werkzeug.utils import redirect
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.http_routing.models.ir_http import slug
+
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteEventMeetController(http.Controller):
+    @http.route(
+        ["/event/<model('event.event'):event>/meeting_rooms"],
+        type="http",
+        auth="public",
+        website=True,
+        sitemap=True,
+    )
+    def meeting_rooms(self, event, lang=None, open_room=None):
+        if not event.can_access_from_current_website():
+            raise NotFound()
+
+        meeting_rooms = event.sudo().meeting_room_ids
+
+        if lang is not None:
+            lang = request.env["res.lang"].browse(int(lang))
+            meeting_rooms = meeting_rooms.filtered(lambda m: m.lang_id == lang)
+
+        values = {
+            "event": event.sudo(),
+            "meeting_rooms": meeting_rooms,
+            "current_lang": lang,
+            "available_languages": event.sudo().meeting_room_ids.mapped("lang_id"),
+            "open_room": int(open_room) if open_room else None,
+            "is_event_manager": request.env.user.has_group("event.group_event_manager"),
+        }
+
+        return request.render("website_event_meet.template_meeting_rooms", values)
+
+    @http.route(["/event/create_meeting_room"], type="http", auth="public", methods=["POST"])
+    def create_meeting_room(self):
+        name = http.request.params.get("name")
+        summary = http.request.params.get("summary")
+        target_audience = http.request.params.get("audience")
+        lang = http.request.params.get("lang")
+        max_capacity = http.request.params.get("capacity")
+        event_id = int(http.request.params.get("event"))
+
+        # get the record to be sure they really exist
+        event = request.env["event.event"].browse(event_id).exists()
+        lang = request.env["res.lang"].search([("code", "=", lang)], limit=1)
+
+        if not event or not event.website_published or not lang:
+            raise Forbidden()
+
+        _logger.info("New meeting room (%s) create by %s" % (name, request.httprequest.remote_addr))
+
+        meeting_room = request.env["event.meeting_room"].sudo().create(
+            {
+                "name": name,
+                "summary": summary,
+                "target_audience": target_audience,
+                "max_capacity": max_capacity,
+                "lang_id": lang.id,
+                "pinned": False,
+                "event_id": event.id,
+            },
+        )
+
+        return redirect("/event/%s/meeting_rooms?open_room=%i" % (slug(event), meeting_room.id))
+
+    @http.route(["/event/update_participant_count"], type="json", auth="public")
+    def update_participant_count(self, meeting_room_id, joined, participant_count):
```

Comment:
we should only update if the given participant_count != participant_count in the table, to avoid too much DB write

Reply:
@dbeguin But we also need to write the `last_joined` column, so we will always lock the row even if the participant count is the same